src/main/java/org/folio/rest/impl/EholdingsResourcesImpl.java

```diff
@@ -0,0 +1,154 @@
+import io.vertx.core.Handler;
+import io.vertx.core.logging.Logger;
+import io.vertx.core.logging.LoggerFactory;
+
+
+public class EholdingsResourcesImpl implements EholdingsResources{
+  
+  private static final String RESOURCE_ID_REGEX = "([^-]+)-([^-]+)-([^-]+)";
+  private static final Pattern RESOURCE_ID_PATTERN = Pattern.compile(RESOURCE_ID_REGEX);
+  private static final String RESOURCE_ID_INVALID_ERROR = "Resource id is invalid";
+  private static final String RESOURCE_NOT_FOUND_MESSAGE = "Resource not found";
+  private static final String CONTENT_TYPE_HEADER = "Content-Type";
+  private static final String CONTENT_TYPE_VALUE = "application/vnd.api+json";
+  
+  private final Logger logger = LoggerFactory.getLogger(EholdingsResourcesImpl.class);
+  
+  private RMAPIConfigurationService configurationService;
+  private HeaderValidator headerValidator;
+  private ResourcesConverter converter;
+  
+  public EholdingsResourcesImpl() {
+    this(
+      new RMAPIConfigurationServiceCache(
+        new RMAPIConfigurationServiceImpl(new ConfigurationClientProvider())),
+      new HeaderValidator(),
+      new ResourcesConverter());
+  }
+  
+  public EholdingsResourcesImpl(RMAPIConfigurationService configurationService,
+      HeaderValidator headerValidator,
+      ResourcesConverter converter) {
+    this.configurationService = configurationService;
+    this.headerValidator = headerValidator;
+    this.converter = converter;
+}
+
+  @Override
+  public void postEholdingsResources(String contentType, ResourcePostRequest entity, Map<String, String> okapiHeaders,
+      Handler<AsyncResult<Response>> asyncResultHandler, Context vertxContext) {
+    asyncResultHandler.handle(Future.succeededFuture(PostEholdingsResourcesResponse.status(Response.Status.NOT_IMPLEMENTED).build()));
+    
+  }
+
+  @Override
+  @HandleValidationErrors
+  public void getEholdingsResourcesByResourceId(String resourceId, String include, Map<String, String> okapiHeaders,
+      Handler<AsyncResult<Response>> asyncResultHandler, Context vertxContext) {
+    ResourceId parsedResourceId = parseResourceId(resourceId);
+    headerValidator.validate(okapiHeaders);
+    
+    CompletableFuture.completedFuture(null)
+      .thenCompose(o -> configurationService.retrieveConfiguration(new OkapiData(okapiHeaders)))
+      .thenCompose(rmapiConfiguration -> {
+        RMAPIService rmapiService = new RMAPIService(rmapiConfiguration.getCustomerId(), rmapiConfiguration.getAPIKey(),
+          rmapiConfiguration.getUrl(), vertxContext.owner());
+        return rmapiService.retrieveResource(parsedResourceId);
+    })
+    .thenAccept(title ->
+      asyncResultHandler.handle(Future.succeededFuture(GetEholdingsResourcesByResourceIdResponse
+        .respond200WithApplicationVndApiJson(converter.convertFromRMAPIResource(title)))))
+    .exceptionally(e -> {
+      logger.error(INTERNAL_SERVER_ERROR, e);
+      if (e.getCause() instanceof RMAPIResourceNotFoundException) {
```

Comment:
Eventually this can be updated to use ErrorHandler from - but probably to be coordinated 
 on after https://github.com/folio-org/mod-kb-ebsco-java/pull/45

Reply:
Yes - agreed! Thanks Carole!